Docs: Update docs for image

ogoffart opened the pull request:
Link to the API docs from the image type.

<!--
- [ ] If the change modifies a visible behavior, it changes the documentation accordingly
- [ ] If possible, the change is auto-tested
- [ ] If the changes fixes or close an existing issue, the commit message reference the issue with `Fixes #xxx` or `Closes #xxx`
- [ ] If the change is noteworthy, the commit message should contain `ChangeLog: ...`
-->

NigelBreslaw reviewed (commented):
There are a couple of open github issues asking for a list of all supported image formats.

tronical commented on docs/astro/src/content/docs/reference/primitive-types.mdx:
```diff
@@ -115,6 +117,24 @@ Slint looks for images in the following places:
+<TabItem label="Rust">
+In Rust, properties or struct fields of the image type are mapped to [`slint::Image`](https://docs.slint.dev/latest/docs/rust/slint/struct.Image).
+
+Some format might be disabled by not enabling the `image-default-formats` cargo feature which is enabled by default.
```
This is a double-negative ("might be disabled by not enabling"). I suggest to simplify this:

```suggestion
Enable the `image-default-formats` feature if you're loading images other than PNG and JPG.
```

ogoffart replied:
Adding SVG and a note that it is the default:

```suggestion
Enable the `image-default-formats` feature (enabled by default) if you're loading images other than PNG, JPEG, and SVG.
```

tronical commented on internal/core/graphics/image.rs:
```diff
@@ -663,7 +663,11 @@ pub struct Image(ImageInner);
+    ///
+    /// Supported format are SVG, as well as format supported by the [`image` crate](https://crates.io/crates/image):
+    /// AVIF, BMP, DDS, Farbfeld, GIF, HDR, ICO, JPEG, EXR, PNG, PNM, QOI, TGA, TIFF, WebP.
+    /// Some format might be disabled by not enabling the `image-default-formats` cargo feature.
```
```suggestion
    /// Enable the `image-default-formats` feature if you're loading images other than PNG and JPG.
```

ogoffart replied:
```suggestion
    /// Enable the `image-default-formats` feature (enabled by default) if you're loading images other than PNG, JPEG, and SVG.
```

ogoffart commented:
I just moved this list in the `image` type where it belongs rather than in the `Image` element.
I removed the extension since the image crate can detect the image format depending on its content and not the extension.

tronical commented on docs/astro/src/content/docs/reference/primitive-types.mdx:
```diff
@@ -115,6 +117,24 @@ Slint looks for images in the following places:
+<TabItem label="Rust">
+In Rust, properties or struct fields of the image type are mapped to [`slint::Image`](https://docs.slint.dev/latest/docs/rust/slint/struct.Image).
+
+Enable the `image-default-formats` feature (enabled by default) if you're loading images other than PNG, JPEG, and SVG.
```
Ah, I keep forgetting about it being default - somehow I thought it isn't. I think advice to enable something that's enabled by default is of no use. I'd rather omit this sentence and we have an "optimzation" section (how to reduce the binary size, speed up compilation), where we can explain that this could be disabled. But in the most prominent place here I think it's confusing.

```suggestion
```

tronical commented on internal/core/graphics/image.rs:
```diff
@@ -663,7 +663,11 @@ pub struct Image(ImageInner);
+    ///
+    /// Supported format are SVG, as well as format supported by the [`image` crate](https://crates.io/crates/image):
+    /// AVIF, BMP, DDS, Farbfeld, GIF, HDR, ICO, JPEG, EXR, PNG, PNM, QOI, TGA, TIFF, WebP.
+    /// Enable the `image-default-formats` feature (enabled by default) if you're loading images other than PNG, JPEG, and SVG.
```
```suggestion
```
